### PR TITLE
add category model

### PR DESCRIPTION
**Pull request by oleksii-sukhenko**

### CODE REVIEW

File: src/main/java/mate/academy/model/Category.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Table(name = "categories")
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@SQLRestriction(value = "is_deleted=false")
+@NoArgsConstructor
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Category.java]:
```suggestion
```

File: src/main/java/mate/academy/model/User.java [hunk 2]
```diff
@@ -34,7 +35,11 @@ public class User implements UserDetails {
     @Column(nullable = false)
     private String lastName;
     private String shippingAddress;
-    @ManyToMany
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/User.java]:
```suggestion
    @ManyToMany
```
you can solve it by using EntityGraph
Just add 

```
@EntityGraph(attributePaths = "roles")
Optional<User> findByEmail(String email);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments
if you have questions, please ask in chat

File: src/main/java/mate/academy/model/User.java (hunk 2 above)
Comment (oleksii-sukhenko) [reply src/main/java/mate/academy/model/User.java]: thanks

File: src/main/java/mate/academy/model/Book.java [hunk 3]
```diff
@@ -16,8 +22,9 @@
 @Setter
 @Entity
 @SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
-@SQLRestriction(value = "is_deleted=false")
+@SQLRestriction(value = "is_deleted = false")
 @Table(name = "books")
+@NoArgsConstructor
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Book.java]: as I can see you don't have any constuctor with params, so you can remove this annotation

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one comment

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**